lib/manager/terraform/artifacts.ts

```diff
@@ -0,0 +1,78 @@
+import { logger } from '../../logger';
+import { ExecOptions, exec } from '../../util/exec';
+import {
+  deleteLocalFile,
+  getSiblingFileName,
+  readLocalFile,
+  writeLocalFile,
+} from '../../util/fs';
+import { UpdateArtifact, UpdateArtifactsResult } from '../common';
+
+const lockFile = '.terraform.lock.hcl';
+
+async function terraformLockUpdate(): Promise<void> {
+  await deleteLocalFile(lockFile);
```

Comment:
Why always delete? Normally we only do this for lockfile maintenance.

Reply:
Do we know if terraform reuses the existing lock file, e.g. to speed up calculations or (ideally) to keep changes to a minimum where possible?

Reply:
This is based upon this comment https://github.com/hashicorp/terraform/issues/27161#issuecomment-742133251 in an issue which I opened to discuss such a reuse of a enxisting lockfile.     
The workaround is to delete the lockfile and then recreate one with `terraform provider lock`